agent/agent.go

```diff
@@ -0,0 +1,60 @@
+package agent
+
+import (
+	"log"
+	"time"
+
+	"github.com/wavefronthq/go-proxy/api"
+)
+
+// Agent interface.
+type WavefrontAgent interface {
+	InitAgent()
+}
+
+type DefaultAgent struct {
+	AgentID    string
+	ApiService api.WavefrontAPI
+	LocalAgent bool
+	PushAgent  bool
+	Ephemeral  bool
+}
+
+func (a *DefaultAgent) InitAgent() {
+	// fetch configuration once per minute
+	checkinTicker := time.NewTicker(time.Minute * time.Duration(1))
+	go a.checkin(checkinTicker)
+}
+
+func (a *DefaultAgent) checkin(ticker *time.Ticker) {
+	for range ticker.C {
+		log.Println("Fetching configuration")
+		a.doCheckin()
+	}
+}
+
+func (a *DefaultAgent) doCheckin() {
+	agentMetrics, err := buildAgentMetrics()
+	if err != nil {
+		log.Println("buildAgentMetrics error", err)
+		return
+	}
+
+	currentTime := getCurrentTime()
+	agentConfig, err := a.ApiService.Checkin(currentTime, a.LocalAgent, a.PushAgent, a.Ephemeral, agentMetrics)
+	if err != nil {
+		log.Println("Checkin error", err)
```

Comment:
not sure how golang logging works, is there a way to escalate this so it seems more urgent than, for example, "fetching configuration" above? maybe log to a different channel?

Reply:
The default Go log package doesn't support log levels. Most projects appear to use third party pkgs or roll out their own logging framework. We may want to do that down the line.

It is possible to have multiple loggers writing to different output files. But this seems suffice for now.